### ISSUE TITLE
Running into GOPATH Error despite having GOPATH set; nothing in log.

### ISSUE THREAD

**Issue by shurcooL** [sourcegraph/sourcegraph-sublime#20]
This may be me doing something wrong rather than an issue, but I can't seem to figure out what's going wrong.

I've installed using the OS X instructions for Sublime Text 3. When I run it and open a channel, I get the following error:

> **GOPATH Error**
>
> No valid GOPATH found in your shell startup scripts or Sublime settings. Please read the GOPATH section in the Sourcegraph Sublime README to learn how to manually set your GOPATH.

My user settings `Sourcegraph.sublime-settings` file looks like this:

```JSON
{
	"GOPATH": "/gopathworkspace1:/gopathworkspace2:/gopathworkspace3",
	"LOG_LEVEL": 3
}
```

Despite that, the log file `/tmp/sourcegraph-sublime.log` does not exist.

In sublime console, I see something like this, and not much else that is relevant:

```
[open_channel] Opening channel in browser: ['open', 'https://sourcegraph.com/-/channel/Dmitri-9af8c...']
Running shell command: open https://sourcegraph.com/-/channel/Dmitri-9af8c...
Running shell command: pwd
[network] Sending post request params: {"Action": {"Fix": "No valid GOPATH found in your shell startup scripts or Sublime settings. Please read the GOPATH section in the Sourcegraph Sublime README to learn how to manually set your GOPATH.", "Error": "GOPATH Error"}, "CheckForListeners": true}
[network] Sending POST request to URL: https://grpc.sourcegraph.com/.api/channel/Dmitri-9af8...
[network] Server responded with code 200
```

Any ideas on what I'm doing wrong? Thanks!

**Comment by uforic** [comment]
This is due to a check we do to make sure the GOPATH exists, "ls $GOPATH". When GOPATH is multiple directories, we are throwing an error. We need to stop assuming GOPATH is one directory.

**Comment by luttig** [comment]
The best design would keep the GOPATH as a string literal (unchanged from
$GOPATH), and pass it into the `env` argument for any calls to
`subprocess.Popen`. For the scenarios in which we use `GOPATH/bin/...`, we
should really just add GOPATH to a PATH variable and pass that into the
`subprocess.Popen` call.

**Comment by shurcooL** [comment]
> This is due to a check we do to make sure the GOPATH exists, "ls $GOPATH". When GOPATH is multiple directories, we are throwing an error. We need to stop assuming GOPATH is one directory.

Makes sense.

In case it's helpful, here's a [definition of GOPATH env var](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) from Go website:

> The GOPATH environment variable lists places to look for Go code. On Unix, the value is a colon-separated string. On Windows, the value is a semicolon-separated string. On Plan 9, the value is a list.

**Comment by shurcooL** [comment]
This has been resolved via #25.
